eldritch/items/unique.py

```diff
@@ -56,3 +58,228 @@ def get_trigger(self, event, owner, state):
+    # TODO: Pass combat check or gate close check
+    if (event.check_type == "combat"
+        or (event.check_type in ("fight", "lore")
+            and isinstance(state.event_stack[-2], events.GateCloseAttempt))):
+      return events.Sequence(
+          [events.DiscardSpecific(owner, [self]), events.PassCheck(owner, event)],
+          owner
+      )
+    return None
+
+
+class EnchantedJewelry(Item):
+  def __init__(self, idx):
+    super().__init__("Enchanted Jewelry", idx, "unique", {}, {}, None, 3)
+    self.tokens["stamina"] = 0
+    self.max_tokens["stamina"] = 3
+
+  def get_usable_interrupt(self, event, owner, state):
+    if not isinstance(event, events.GainOrLoss):
+      return None
+
+    if not owner == event.character:
+      return None
+
+    loss = event.losses.get("stamina", 0)
+    if isinstance(loss, values.Value):
+      loss = loss.value(state)
+
+    if loss <= 0:
+      return None
+
+    reduction = events.LossPrevention(self, event, "stamina", 1)
+    return events.Sequence(
+        [
+            reduction,
+            events.AddToken(self, "stamina", event.character, n_tokens=1)
+        ],
+        owner
+    )
+
+
+class GateBox(Item):
+  def __init__(self, idx):
+    super().__init__("Gate Box", idx, "unique", {}, {}, None, 4)
+
+  def get_interrupt(self, event, owner, state):
+    if not isinstance(event, events.GateChoice) or not event.character == owner:
+      return None
+    if event.gate_name is None:
+      return None
+    return events.Sequence([
+        events.CancelEvent(event),
+        events.GateChoice(owner, "Choose any open gate", None, event.none_choice, event.annotation)
+    ],
+        owner)
+
+
+class HealingStone(Item):
+  def __init__(self, idx):
+    super().__init__("Healing Stone", idx, "unique", {}, {}, None, 8)
+
+  # Copied get_usable{_trigger,_interrupt,} from Physician ability
+  def get_usable_trigger(self, event, owner, state):
+    if not isinstance(event, events.UpkeepActions):
+      return None
+    return self.get_usable(event, owner, state)
+
+  def get_usable_interrupt(self, event, owner, state):
+    if not isinstance(event, (events.UpkeepActions, events.SliderInput)):
+      return None
+    return self.get_usable(event, owner, state)
+
+  def get_usable(self, event, owner, state):
+    if self.exhausted:
+      return None
+    if event.character != owner:
+      return None
+    # End copy-paste
+
+    available = [attr for attr in ["stamina", "sanity"] if getattr(owner, attr) < getattr(owner, f"max_{attr}")(state)]
+    if not available:
+      return None
+    elif len(available) == 1:
+      gain = events.Gain(owner, {available[0]: 1})
+    else:
+      gain = events.BinaryChoice(
+          owner,
+          "Gain 1 Stamina or 1 Sanity?",
+          "1 Stamina",
+          "1 Sanity",
+          events.Gain(owner, {"stamina": 1}),
+          events.Gain(owner, {"sanity": 1}),
+      )
+    # While the text doesn't explicitly say to exhaust, if you don't,
```

Comment:
Thinking about this, maybe it would make sense to turn this into a trigger instead of a usable trigger (with the option to gain nothing). Although, unlike the Ruby, it can't be traded before someone else can use it (since trading only happens during movement). Maybe I'm overthinking it. If we find some way for players to trade during upkeep, the fact that it technically doesn't exhaust could be important.